### PR TITLE
feat: editor page

### PR DESCRIPTION
**Pull request by tenadopt**
reset to the basic. Start from the beginning. Observe the work.

### CODE REVIEW

File: src/app/ErrorBoundary.tsx
Comment (shev7) [inline src/app/ErrorBoundary.tsx:1]: for catching errors in the application you can use such feature of react-router-dom as errorElement
https://reactrouter.com/en/main/route/error-element

File: src/app/routes.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import ImageEditorPage from 'pages/ImageEditor/ui/ImageEditorPage';
+import AppLayout from 'widgets/AppLayout';
+
+const AppRoutes = () => (
+    <Routes>
+        <Route path="/" element={<AppLayout />}>
+            <Route path="editor" element={<ImageEditorPage />} />
+        </Route>
+    </Routes>
+);
```
Comment (shev7) [inline src/app/routes.tsx]: you should use createBrowserRouter

File: src/features/imageEditor/ui/ImageEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React, { useState, Suspense, lazy } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "app/store";
+import { Button, Slider, TextField } from "@mui/material";
+import { setBrushSizeWithState, setColorWithState } from "features/imageEditor/model/imageEditorSlice";
+
+// Example of dynamic import (lazy loading)
+const LargeComponent = lazy(() => import('./LargeComponent'));
```
Comment (shev7) [inline src/features/imageEditor/ui/ImageEditor.tsx]: you should apply lazy loading for pages

File: src/widgets/Header/ui/Header.tsx
Comment (shev7) [inline src/widgets/Header/ui/Header.tsx:1]: why are you creating ButtonLink component and then using regular button

File: webpack.config.ts [hunk 3]
```diff
@@ -8,13 +8,22 @@ interface Configuration extends WebpackConfiguration {
...
         extensions: ['.ts', '.tsx', '.js'],
+        alias: {
+            'app': path.resolve(__dirname, 'src/app'),
+            'entities': path.resolve(__dirname, 'src/entities'),
+            'features': path.resolve(__dirname, 'src/features'),
+            'shared': path.resolve(__dirname, 'src/shared'),
+            'pages': path.resolve(__dirname, 'src/pages'),
+            'widgets': path.resolve(__dirname, 'src/widgets'),
+        },
     },
```
Comment (shev7) [inline webpack.config.ts]: for this purpose you can use tsconfig paths plugin

**Review by shev7 — CHANGES_REQUESTED** [review]

File: src/app/index.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+const container = document.getElementById('root');
+const root = createRoot(container!);
+
+root.render(
+    <React.StrictMode>
+        <Provider store={store}>
+            <BrowserRouter>
+                <CssBaseline />
+                <App />
```
Comment (GoodPassiveMan) [inline src/app/index.tsx]: I see this twice in your code. Is it ok?

```
<CssBaseline />
<App />
```

**Review by GoodPassiveMan — CHANGES_REQUESTED** [review]
Base this branch on branch from previous PR as them both have similar code. So I won't need to check same code twice

File: src/app/ErrorBoundary.tsx
Comment (tenadopt) [reply src/app/ErrorBoundary.tsx:1]: done

File: src/app/routes.tsx (hunk 1 above)
Comment (tenadopt) [reply src/app/routes.tsx]: done

File: src/features/imageEditor/ui/ImageEditor.tsx (hunk 2 above)
Comment (tenadopt) [reply src/features/imageEditor/ui/ImageEditor.tsx]: done

File: src/widgets/Header/ui/Header.tsx
Comment (tenadopt) [reply src/widgets/Header/ui/Header.tsx:1]: replace for the useNavigate

File: src/app/index.tsx (hunk 4 above)
Comment (tenadopt) [reply src/app/index.tsx]: Update

File: webpack.config.ts (hunk 3 above)
Comment (tenadopt) [reply webpack.config.ts]: Done

File: src/features/imageEditor/ui/CanvasEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+        context.beginPath();
+        context.rect(x1, y1, width, height);
+        context.stroke();
+    };
+
+    const handleBrushSizeChange = (_: Event, newValue: number | number[]) => {
+        if (typeof newValue === 'number') {
+            setBrushSizeState(newValue);
+            dispatch(setBrushSizeWithState({ brushSize: newValue }));
```
Comment (GoodPassiveMan) [inline src/features/imageEditor/ui/CanvasEditor.tsx:151]: Why would you need two states for same thing?

File: src/features/imageEditor/ui/CanvasEditor.tsx [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+        }
+    };
+
+    const handleColorChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setColorState(event.target.value);
+        dispatch(setColorWithState({ color: event.target.value }));
```
Comment (GoodPassiveMan) [inline src/features/imageEditor/ui/CanvasEditor.tsx:156]: Same here. 2 identical states

**Review by GoodPassiveMan — CHANGES_REQUESTED** [review]
You don't reset brush size after you leave canvas. Does not seem right to me as it's reset after page reload but not page leave

**Comment by tenadopt** [comment]
> Base this branch on branch from previous PR as them both have similar code. So I won't need to check same code twice

Solve it.

File: src/features/imageEditor/ui/CanvasEditor.tsx (hunk 5 above)
Comment (tenadopt) [reply src/features/imageEditor/ui/CanvasEditor.tsx:151]: You were right! Fix it!

File: src/features/imageEditor/ui/CanvasEditor.tsx (hunk 6 above)
Comment (tenadopt) [reply src/features/imageEditor/ui/CanvasEditor.tsx:156]: You were right! Fix it!

**Comment by tenadopt** [comment]
> Base this branch on branch from previous PR as them both have similar code. So I won't need to check same code twice



> You don't reset brush size after you leave canvas. Does not seem right to me as it's reset after page reload but not page leave

**Comment by tenadopt** [comment]
> You don't reset brush size after you leave canvas. Does not seem right to me as it's reset after page reload but not page leave

Solve it

**Review by GoodPassiveMan — APPROVED** [review]

**Pull request merged**